src/seedu/addressbook/commands/assessment/AddGradesCommand.java

```diff
@@ -0,0 +1,90 @@
+package seedu.addressbook.commands.assessment;
+
+import seedu.addressbook.commands.Command;
+import seedu.addressbook.commands.commandresult.CommandResult;
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Assessment;
+import seedu.addressbook.data.person.Grades;
+import seedu.addressbook.data.person.Person;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+
+/**
+ * Adds a grade to person based on person index and assessment index.
+ */
+public class AddGradesCommand extends Command {
+
+    public static final String COMMAND_WORD = "addgrades";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ":\n" + "Adds grades to a person based on person index"
+            + " and grades index.\n\t"
+            + "Parameters: PERSON_INDEX ASSESSMENT_INDEX Grade\n\t"
+            + "Example: " + COMMAND_WORD
+            + " 1 2 85";
+
+    public static final String MESSAGE_ADD_GRADE_SUCCESS = "Grade has been added!";
+
+    private final int assessIndex;
+    private final int gradesVal;
+
+    /**
+     * Constructor used for Privileges
+     * Command constructed has no functionality
+     * */
+    public AddGradesCommand() {
+        this.assessIndex = -1;
+        this.gradesVal = -1;
+    }
+
+    /**
+     * Convenience constructor using raw values.
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+    public AddGradesCommand (int targetPersonIndex, int targetAssessIndex, int gradesVal) {
+        super (targetPersonIndex);
+        assessIndex = targetAssessIndex;
+        this.gradesVal = gradesVal;
+    }
+
+    @Override
+    public CommandResult execute() {
+        try {
+            final ReadOnlyPerson target = getTargetPerson();
+            final Person person = addressBook.findPerson(target);
+
```

Comment:
I think with that recent patch, getTargetPerson returns a (editable) Person, while getTargetReadOnlyPerson returns uneditable one. 
So you don't need both function calls to get the person.

Reply:
He means just use Person person = getTargetPerson();